python/ql/src/semmle/python/dataflow/Implementation.qll

```diff
@@ -726,6 +726,8 @@ private class EssaTaintTracking extends string {
     private TaintKind iterable_unpacking_descent(
         SequenceNode left_parent, ControlFlowNode left_defn, CollectionKind parent_kind
     ) {
+        //TODO: Fix the cartesian product in this predicate
+        none() and
```

Comment:
Is this necessary?

Reply:
More or less. If it isn't there, the cartesian product will get evaluated. With this `none`, the predicate becomes empty, as it has no base case.

I could also have just removed the predicate entirely, but I'm hoping there's a way to fix it up so that we avoid the CP. Either way, this is only intended to be a short-term fix.

Also, in case you're wondering why the CP appears in the first place, it's because there's no relationship between `parent_kind` and the remaining arguments. That means the base case has to hold for _all_ combinations of potential `CollectionKind`s.

Reply:
I suppose what I was thinking was, why not comment out the remainder of the body, but this works just as well. :)